miden-lib/asm/sat/asset.masm

```diff
@@ -0,0 +1,83 @@
+use.miden::sat::internal::account
+use.miden::sat::internal::asset
+
+# CONSTANTS
+# =================================================================================================
+
+#! Builds a fungible asset for the specified fungible faucet and amount.
+#!
+#! Stack: [faucet_id, amount]
+#! Output: [ASSET]
+#!
+#! - faucet_id is the faucet to create the asset for.
+#! - amount is the amount of the asset to create.
+#! - ASSET is the built fungible asset.
+export.build_fungible_asset
+    # assert the faucet is a fungible faucet
+    dup exec.account::is_fungible_faucet assert
+    # => [faucet_id, amount]
+
+    # assert the amount is valid
+    dup.1 exec.asset::get_fungible_asset_max_amount lte assert
+    # => [faucet_id, amount]
+
+    # create the asset
+    push.0 swap push.0 swap
+    # => [ASSET]
+end
+
+#! Creates a fungible asset for the faucet the transaction is being executed against.
+#!
+#! Stack: [amount]
+#! Output: [ASSET]
+#!
+#! - amount is the amount of the asset to create.
+#! - ASSET is the created fungible asset.
+export.create_fungible_asset
+    # fetch the id of the faucet the transaction is being executed against.
+    exec.account::get_id
+    # => [id, amount]
```

Comment:
Since this is intended to be called from the user context, should this be a `syscall`? Otherwise, `account::get_id` would be executed against non-root context's memory and would return an incorrect result.

Reply:
great catch.